### ISSUE TITLE
[Feature Request] Support passing 4 values to padding for Conv2dOp

### ISSUE THREAD

**Issue by jserbedzijaTT** [tenstorrent/tt-metal#17656]
**Is your feature request related to a problem? Please describe.**
In TT MLIR, we encountered this issue: https://github.com/tenstorrent/tt-mlir/issues/1373, where Padding Top and Padding Bottom had different values, which is currently not supported in TTNN.

**Describe the solution you'd like**
Support passing four values to padding for Conv2dOp: [padding_top, padding_left, padding_bottom, padding_right].

**Comment by pavlejosipovic** [comment]
@jserbedzijaTT @sdjordjevicTT This seems to come from some pytorch model right?
As far as I know pytorch doesn't support this flexibility (having different amount of padding for top/bottom).
https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html

In ttnn we try to stick to torch interface as far as we can.
I'm curious how did you get this situation, am I missing something?

Is using separate ttnn::pad op an option for compiler in these situations?

**Comment by sdjordjevicTT** [comment]
@ashokkumarkannan1 FYI as you hit and reported a similar issue on tt-mlir:
https://github.com/tenstorrent/tt-mlir/issues/2200

@ashokkumarkannan1 can you comment more about from which model and frontend you hit this issue?

**Comment by ashokkumarkannan1** [comment]
Hi @sdjordjevicTT  
This  issue is faced in the mobilenet v1 and v2  pytorch models.
cc: @nvukobratTT

**Comment by ashokkumarkannan1** [comment]
Hi @nvukobratTT ,

> 1. Are inputs in the expected shape (not something strange, but rather something that the customer would use as well)

Yes inputs are in expected shape.

> 2. Do we have a bug around padding setting/calculation (e.g. invalid way of setting 4-digit padding, or rounding error)

Yes. We have. When we have `padding_top != padding_bottom` or `padding_left != padding_right` we face this issue. As a summary, ttnn.Conv2d currently support only uniform padding. It expects `padding_top == padding_bottom` and `padding_left == padding_right`.

> 3. Do we fuse padding in conv2d op (if yes, with new backend we might want to split those up)

Yes. We are doing it here. - https://github.com/tenstorrent/tt-tvm/blob/690fb995784df59635a399a731c7e1b1394bfb8d/python/tvm/relay/op/contrib/forge/forge_passes.py#L138
To fix this we have to retain padding in asymmetric cases, and we can map those pad op to [ttir.PadOp](https://docs.tenstorrent.com/tt-mlir/autogen/md/Dialect/TTIROp.html#ttirpad-ttttirpadop).  We can fuse only for symmetric cases.

Shall we proceed with this fix @nvukobratTT ?

**Comment by nvukobratTT** [comment]
@ashokkumarkannan1 yup, let's proceed with fix, and skip fusing in cases of asymmetric padding. In others, let's continue with fusing as before.

**Comment by nvukobratTT** [comment]
@sdjordjevicTT @pavlejosipovic we can close this issue as we'll handle this one FFE side.

**Comment by pavlejosipovic** [comment]
Hi @nvukobratTT @jserbedzijaTT This feature got implemented here 
https://github.com/tenstorrent/tt-metal/issues/18918
